build.gradle

```
@@ -1,76 +1,87 @@
 plugins {
-	id 'org.quiltmc.loom' version '0.12-SNAPSHOT'
 	id 'maven-publish'
+	alias(libs.plugins.quilt.loom)
 }
 
-group = maven_group
+archivesBaseName = project.archives_base_name
 version = project.version
-archivesBaseName = archives_base_name
+group = project.maven_group
 
 repositories {
+	// Add repositories to retrieve artifacts from in here.
+	// You should only use this when depending on other mods because
+	// Loom adds the essential maven repositories to download Minecraft and libraries from automatically.
+	// See https://docs.gradle.org/current/userguide/declaring_repositories.html
+	// for more information about repositories.
 }
 
+// All the dependencies are declared at "./gradle/libs.version.toml" and referenced with "libs.<id>"
+// See https://docs.gradle.org/current/userguide/platforms.html for information on how version catalogs work.
 dependencies {
-	// To change the versions see the gradle.properties file
-	minecraft "com.mojang:minecraft:$minecraft_version"
-	mappings( loom.layered {
-		addLayer quiltMappings.mappings( "org.quiltmc:quilt-mappings:$minecraft_version+build.$quilt_mappings:v2" )
-//		addLayer loom.officialMojangMappings() // uncomment if you want mojmap too
-	})
-	modImplementation "org.quiltmc:quilt-loader:$loader_version"
-
-	modImplementation "org.quiltmc:qsl:$qsl_version"
+	minecraft libs.minecraft
+	mappings loom.layered {
+		addLayer quiltMappings.mappings("org.quiltmc:quilt-mappings:${libs.versions.quilt.mappings.get()}:v2")
+		// addLayer loom.officialMojangMappings() // Uncomment if you want to use Mojang Mappings as your primary mappings, falling back on QM for parameters and Javadocs
```

```suggestion
		// officialMojangMappings() // Uncomment if you want to use Mojang Mappings as your primary mappings, falling back on QM for parameters and Javadocs
```
 loom doesn't like you doing it that way